modules/planning/tasks/deciders/open_space_roi_decider.cc

```diff
@@ -324,59 +324,61 @@ bool OpenSpaceRoiDecider::GetOpenSpaceROI() {
 
   // get end_pose of the parking spot
   parking_spot_heading_ = (left_down - left_top).Angle();
-  double end_x = (left_top.x() + right_top.x()) / 2;
+  double end_x = (left_top.x() + right_top.x()) / 2.0;
   double end_y = 0.0;
+  const double parking_depth_buffer =
+      config_.open_space_roi_decider_config().parking_depth_buffer();
+  CHECK_GE(parking_depth_buffer, 0.0);
+  const bool parking_inwards =
+      config_.open_space_roi_decider_config().parking_inwards();
   if (parking_spot_heading_ > kMathEpsilon) {
-    if (config_.open_space_roi_decider_config().parking_inwards()) {
-      end_y = left_down.y() - std::max(3 * (left_down.y() - left_top.y()) / 4,
-                                       vehicle_params_.front_edge_to_center());
+    if (parking_inwards) {
+      end_y =
+          left_down.y() - (std::max(3 * (left_down.y() - left_top.y()) / 4,
+                                    vehicle_params_.front_edge_to_center()) +
+                           parking_depth_buffer);
+
     } else {
-      end_y = left_down.y() - std::max((left_down.y() - left_top.y()) / 4,
-                                       vehicle_params_.back_edge_to_center());
+      end_y = left_down.y() - (std::max((left_down.y() - left_top.y()) / 4,
+                                        vehicle_params_.back_edge_to_center()) +
+                               parking_depth_buffer);
     }
   } else {
-    if (config_.open_space_roi_decider_config().parking_inwards()) {
-      end_y = left_down.y() + std::max(3 * (left_top.y() - left_down.y()) / 4,
-                                       vehicle_params_.front_edge_to_center());
+    if (parking_inwards) {
+      end_y =
+          left_down.y() + (std::max(3 * (left_top.y() - left_down.y()) / 4,
+                                    vehicle_params_.front_edge_to_center()) +
+                           parking_depth_buffer);
     } else {
-      end_y = left_down.y() + std::max((left_top.y() - left_down.y()) / 4,
-                                       vehicle_params_.back_edge_to_center());
+      end_y = left_down.y() + (std::max((left_top.y() - left_down.y()) / 4,
+                                        vehicle_params_.back_edge_to_center()) +
+                               parking_depth_buffer);
```

Comment:
Try to avoid complicate multi-lines statements.

Reply:
fixed